File: backend/src/main/java/com/synapse/clinicafemina/domain/Agendamento.java
package com.synapse.clinicafemina.domain;

import com.synapse.clinicafemina.integration.external.ExternalProviderType;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.OffsetDateTime;

@Entity
@Table(name = "agendamento")
@Getter
@Setter
public class Agendamento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "clinica_id", nullable = false)
    private Clinica clinica;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "paciente_id", nullable = false)
    private Paciente paciente;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "medico_id")
    private Usuario medico;

    @Enumerated(EnumType.STRING)
    @Column(name = "external_source", nullable = false, length = 20)
    private ExternalProviderType externalSource;

    @Column(name = "external_id", nullable = false, length = 100)
    private String externalId;

    @Column(name = "data_hora_inicio", nullable = false)
    private OffsetDateTime dataHoraInicio;

    @Column(name = "data_hora_fim")
    private OffsetDateTime dataHoraFim;

    @Column(length = 40)
    private String tipo;

    @Column(name = "servico_nome", length = 120)
    private String servicoNome;

    @Column(nullable = false, length = 30)
    private String status = "AGENDADO";

    @Column(length = 40)
    private String origem = "INTEGRACAO_EXTERNA";

    @Column(name = "confirmado_em")
    private OffsetDateTime confirmadoEm;

    @Column(name = "cancelado_em")
    private OffsetDateTime canceladoEm;

    @Column(name = "motivo_cancelamento", length = 255)
    private String motivoCancelamento;

    @Column(name = "confirmacao_enviada")
    private Integer confirmacaoEnviada;

    @Column(name = "external_payload", columnDefinition = "jsonb")
    @JdbcTypeCode(SqlTypes.JSON)
    private String externalPayload;

    @Column(name = "criado_em", nullable = false, updatable = false)
    private OffsetDateTime criadoEm;

    @Column(name = "atualizado_em", nullable = false)
    private OffsetDateTime atualizadoEm;

    @PrePersist
    protected void onCreate() {
        criadoEm = OffsetDateTime.now();
        atualizadoEm = OffsetDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        atualizadoEm = OffsetDateTime.now();
    }
}

File: frontend/src/types/agendamento.ts
export type Agendamento = {
  id: number;
  pacienteId: number;
  pacienteNome: string;
  medicoId: number | null;
  medicoNome: string | null;
  medicoExternalId: string | null;
  medicoOrigem: string | null;
  dataHoraInicio: string;
  dataHoraFim: string | null;
  tipo: string;
  servicoNome: string;
  status: string;
  origem: string;
  confirmacaoEnviada: number | null;
  canceladoEm: string | null;
  motivoCancelamento: string | null;
};

export type AgendaOption = {
  id: number | null;
  nome: string;
  codigoExterno: string | null;
  origem: string | null;
};

export type AgendaOptions = {
  pacientes: AgendaOption[];
  medicos: AgendaOption[];
};

export type AgendamentoPayload = {
  pacienteId: number;
  medicoId: number | null;
  dataHoraInicio: string;
  dataHoraFim: string | null;
  tipo: string;
  servicoNome: string;
};
